geopandas/geoseries.py

```diff
@@ -170,7 +168,10 @@ def _wrapped_pandas_method(self, mtd, *args, **kwargs):
 
     def __getitem__(self, key):
         if isinstance(key, (slice, list, Series, np.ndarray)):
-            return super(GeoSeries, self).__getitem__(key)
+            block = self._data._block[key]
+            index = self.index[key]
+            return GeoSeries(SingleBlockManager(block, axis=index),
```

Comment:
yep, the getitem on a GeoSeries was still failing (at least slicing was failing). I was looking how it could be fixed in pandas, but in the meantime probably good to fix it here.